docs/models/advanced/extended-blockstates.md

```diff
@@ -0,0 +1,19 @@
+Extended Blockstates
+====================
+
+Extended blockstates provide a way for blocks to pass arbitrary data to their models. Ordinary blockstates and properties occupy only a fixed set of possible states, while extended states can form infinite sets. This accomplished through the use of unlisted properties (`IUnlistedProperty<V>`), extended block states (`IExtendedBlockState`), and extended block state containers (`ExtendedBlockState`).
+
+Ordinary block state containers (`BlockStateContainer`) take a set of listed properties (`IProperty<V>`) that define all possible values for themselves and use those to create all possible combinatons of values. These combinations are then turned into `IBlockState`s and stored in the finite set of all possible states. The properties are called "listed" as they appear on the F3 debug screen, all the way to the right.
+
+Extended block state containers take a set of listed properties, and also a set of *unlisted* properties. Unlisted properties do not define all of their possible values, but instead provide a predicate for checking if a value is valid. The listed properties are again used to create the set of states, this time `IExtendedBlockState`s, but the unlisted properties remain unset. The unlisted properties are only set when `IExtendedBlockState::withProperty` is called to do so. This allows unlisted properties to have any type, even other `IBlockState`s.
```

Comment:
yes but just because there's a validator doesn't mean anything in the context of fixed vs unfixed values. I could have a validator on the fixed value set too